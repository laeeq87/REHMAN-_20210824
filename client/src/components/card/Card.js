import React from 'react'
import { Card } from 'react-bootstrap'
import { OverlayTrigger, Tooltip } from "react-bootstrap";

export const VideoCard = (props) => {
    const thumbnailPath = `http://localhost:5000/thumbnails/${props.thumbnail}`;
    return (
        <div onClick={props.onClick} className="mt-3">
        <OverlayTrigger overlay={ <Tooltip className="" id="tooltip-top">{props.title}</Tooltip>} placement={"top"}>
            <Card style={{ width: 350, height: 300, borderRadius: 10 }}>
            <Card.Img style={{ width: 350, height: 200, borderRadius: 10 }} variant="top" src={thumbnailPath} />
            <Card.Body>
                <Card.Title data-testid="tst-title"><strong>{props.title}</strong></Card.Title>
                <Card.Text data-testid="tst-category"><strong>Category: </strong>{props.category}</Card.Text>
            </Card.Body>
            </Card>
        </OverlayTrigger>
    </div>
    )
  };