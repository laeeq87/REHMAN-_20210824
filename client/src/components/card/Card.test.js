import React from "react";
import { render, screen, cleanup} from '@testing-library/react';
import { VideoCard } from "./Card";
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'


afterEach(()=> {
    cleanup();
});

function clickCalled() {
    console.log('Clicked Called');
}

it("Should Render Video Card Component", ()=> {
    const props = {
        thumbnail: "link",
        title: "Video",
        category: 1,
        onClick: clickCalled
    };
    render(<VideoCard props={props} />);
    const videoCardElement = screen.getByTestId('tst-title')
    expect(videoCardElement).toBeInTheDocument();
    expect(videoCardElement).toContainHTML('<strong>');
}); 

test('matches snapshot', ()=> {
    const props = {
        thumbnail: "link",
        title: "Video",
        category: 1,
        onClick: clickCalled
    };

    const tree = renderer.create(<VideoCard props={props} />).toJSON();
    expect(tree).toMatchSnapshot();


})