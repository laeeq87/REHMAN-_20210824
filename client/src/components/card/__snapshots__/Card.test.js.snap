// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-3"
>
  <div
    className="card"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "borderRadius": 10,
        "height": 300,
        "width": 350,
      }
    }
  >
    <img
      className="card-img-top"
      src="http://localhost:5000/thumbnails/undefined"
      style={
        Object {
          "borderRadius": 10,
          "height": 200,
          "width": 350,
        }
      }
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
        data-testid="tst-title"
      >
        <strong />
      </div>
      <p
        className="card-text"
        data-testid="tst-category"
      >
        <strong>
          Category: 
        </strong>
      </p>
    </div>
  </div>
</div>
`;
